homeassistant/components/jellyfin/sensor.py

```diff
@@ -0,0 +1,76 @@
+"""Support for Jellyfin sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DATA_COORDINATORS, DOMAIN
+from .coordinator import JellyfinDataT, JellyfinDataUpdateCoordinator
+from .entity import JellyfinEntity
+
+
+@dataclass
+class JellyfinSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[JellyfinDataT], StateType]
+
+
+@dataclass
+class JellyfinSensorEntityDescription(
+    SensorEntityDescription, JellyfinSensorEntityDescriptionMixin
+):
+    """Describes Jellyfin sensor entity."""
+
+
+def _count_now_playing(data: JellyfinDataT) -> int:
+    """Count the number of now playing."""
+    session_ids = [
+        sid for (sid, session) in data.items() if "NowPlayingItem" in session
+    ]
+
+    return len(session_ids)
+
+
+SENSOR_TYPES: dict[str, JellyfinSensorEntityDescription] = {
+    "sessions": JellyfinSensorEntityDescription(
+        key="watching",
+        icon="mdi:television-play",
+        native_unit_of_measurement="Watching",
+        value_fn=_count_now_playing,
+    )
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Jellyfin sensor based on a config entry."""
+    coordinators: dict[str, JellyfinDataUpdateCoordinator[Any]] = hass.data[DOMAIN][
+        entry.entry_id
+    ][DATA_COORDINATORS]
```

Comment:
It might be a bit nicer to store a dataclass in `hass.data[DOMAIN][entry.entry_id]` if its not too much work to refactor

Reply:
Example in `lookin/models.py`

Reply:
that's a great example thanks.

Reply:
converted over to such a pattern